app/models/reservation.rb

```diff
@@ -63,6 +63,7 @@ def self.joins_order
   scope :not_canceled, -> { where(canceled_at: nil) }
   scope :not_started, -> { where(actual_start_at: nil) }
   scope :not_ended, -> { where(actual_end_at: nil) }
+  scope :not_in_cart, -> { joins(:order).where.not("orders.ordered_at" => nil) }
```

Comment:
Theoretically, we shouldn't need this additional check because `OrderDetail#state` _should_ be null until it's purchased, so the `state: %w(new inprocess)` _should_ be good enough. However, there's a weird case with merge orders that we recently discovered in #731 where merge orders that need attention (need a reservation or missing an order form) have `ordered_at` set and have the `state` set to `new`. In that PR, we get rid of the `ordered_at` setting, but the state piece seemed to complicated to deal with. 

We can probably ignore this edge case because if there's no reservation, we don't need to worry about this hitting it.

There's no harm in doing a check on purchased, so if you do think we want to keep it, I would recommend using the `purchased` state of order, not `ordered_at IS NULL`. (there's already a `purchased` scope on `OrderDetail` that does the `joins(:order).merge(Order.purchased)`. I'd also recommend sticking to `purchased` as the name of the scope if we add a new scope here for consistency.